jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -209,6 +227,17 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 			resultAttribute.addTo(resultTagMap, result);
 		}
 
+		if (typeNamesAsTags) {
+			log.debug("adding typeNames as tags");
+			Map<String, String> typeNameValueMap = TypeNameValue.extractMap(resultTagMap.get("typeName"));
+			log.debug("this is typeNameValueMap {} ==== {} ", typeNameValueMap, this.typeNames);
+			for (String typeToTag : this.typeNames) {
+				if (typeNameValueMap.containsKey(typeToTag)) {
```

Comment:
Replace the `containsKey()` call with a get and check for nullity. This save one lookup in the map.